City-of-Helsinki/drupal-helfi-platform-config
# UHF-9033

# [UHF-9033](https://helsinkisolutionoffice.atlassian.net/browse/UHF-9033)

Käyttäjäkysely-popup

## What was done

Added code changes to allow use of käyttäjäkysely -popup
Added ws-2.3.0.js -library to git so we can be sure that the correct library is always used
Added user inquiry as a leijuke-block option.

## How to install ON ETUSIVU
* Make sure your instance is up and running on latest dev branch.
    * `git pull origin dev`
    * `make fresh`
* Update the Helfi Platform config
    * `composer require drupal/helfi_platform_config:dev-UHF-9033`
* Run `make drush-updb drush-cr`

## How to test

- You should test this on etusivu instance since etusivu is where this will be set up.
- Log in as admin
- Go to block layout and add new leijuke-block to attachment-section
  - Select käyttäjäkysely as a type
  - No need for any restrictions
  - Save the block

- Go to any page as incognito user, Accept all cookies
  - You should see the käyttäjäkysely-popup 
  - Close it or finish answering. it doesn't matter.
  - Reload the page. You shouldn't see the popup any more.
- Close the incognito window and reopen the incognito window, go to any page on etusivu instance (clears the incognito cookies)
  - You should see the cookie banner again, decline all cookies
  - You should not see the popup

ALSO:
Make sure that the old leijuke-chat blocks still have the open button on bottom right corner


[UHF-9033]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-9033?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
